pantalaimon/daemon.py

```diff
@@ -1039,19 +1079,44 @@ async def search(self, request):
 
         return web.json_response(result, headers=CORS_HEADERS, status=200)
 
-    async def download(self, request):
-        server_name = request.match_info["server_name"]
-        media_id = request.match_info["media_id"]
-        file_name = request.match_info.get("file_name")
+    async def upload(self, request):
+        file_name = request.query.get("filename", "")
+        content_type = request.headers.get("Content-Type", "application/octet-stream")
+        client = next(iter(self.pan_clients.values()))
+
+        try:
+            response = await client.upload(
+                data_provider=await request.read,
+                content_type=content_type,
+                filename=file_name,
+                encrypt=True,
+            )
+
+            if not isinstance(response, UploadResponse):
+                return web.Response(
+                    status=response.transport_response.status,
+                    content_type=response.transport_response.content_type,
+                    headers=CORS_HEADERS,
+                    body=await response.transport_response.read(),
+                )
+
+            self.store.save_upload(response.content_uri)
```

Comment:
You aren't storing any keys here, no. How will you be able to decrypt stuff again?